packages/enclave/src/v3/eth/wallets/wallets.service.ts

```diff
@@ -1,4 +1,160 @@
 import { Injectable } from "@nestjs/common";
+import { SDK, Transaction } from "@haechi-labs/henesis-wallet-core";
+import { WalletDTO } from "../dto/wallet.dto";
+import { BalanceDTO } from "../dto/balance.dto";
+import { ChangeWalletNameRequestDTO } from "./dto/change-wallet-name-request.dto";
+import { SendCoinRequestDTO } from "./dto/send-coin-request.dto";
+import { TransactionDTO } from "../dto/transaction.dto";
+import BN from "bn.js";
+import { CreateTransactionRequestDTO } from "./dto/create-transaction-reqeust.dto";
+import { CreateFlushRequestDTO } from "./dto/create-flush-request.dto";
+import { PaginationDTO } from "../dto/pagination.dto";
+import { DepositAddressDTO } from "../dto/deposit-address.dto";
+import { CreateDepositAddressRequestDTO } from "./dto/create-deposit-address-request.dto";
+import {
+  EthWallet,
+  UserWalletPaginationOptions,
+} from "@haechi-labs/henesis-wallet-core/lib/eth/wallet";
 
 @Injectable()
-export class WalletsService {}
+export class WalletsService {
+  public async getWallets(sdk: SDK, name?: string): Promise<WalletDTO[]> {
+    const options = {};
+    if (name != null) {
+      options["name"] = name;
+    }
+    return (await sdk.eth.wallets.getWallets(options)).map(
+      WalletDTO.fromEthWallet
+    );
+  }
+
+  public async getWallet(sdk: SDK, walletId: string): Promise<WalletDTO> {
+    return WalletDTO.fromEthWallet(await sdk.eth.wallets.getWallet(walletId));
+  }
+
+  public async getBalanceOfWallet(
+    sdk: SDK,
+    walletId: string,
+    ticker?: string
+  ): Promise<BalanceDTO[]> {
+    const wallet: EthWallet = await sdk.eth.wallets.getWallet(walletId);
+    return (await wallet.getBalance())
+      .map(BalanceDTO.fromBalance)
+      .filter((balance) => {
+        if (ticker === undefined) return true;
+        return balance.ticker.toUpperCase() == ticker.toUpperCase();
+      });
+  }
+
+  public async changeWalletName(
+    sdk: SDK,
+    walletId: string,
+    request: ChangeWalletNameRequestDTO
+  ) {
+    const wallet: EthWallet = await sdk.eth.wallets.getWallet(walletId);
+    await wallet.changeName(request.name);
+  }
+
+  public async sendCoin(
+    sdk: SDK,
+    walletId: string,
+    request: SendCoinRequestDTO
+  ): Promise<TransactionDTO> {
+    const wallet: EthWallet = await sdk.eth.wallets.getWallet(walletId);
+    return TransactionDTO.fromTransaction(
+      ((await wallet.transfer(
```

Comment:
as unknown 으로 한 이유가 있을까요~?

Reply:
unknown으로 typecase를 하지 않으면 아래와 같은 에러가 발생합니다.

TS2352: Conversion of type 'EthTransaction' to type 'Transaction' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.

그래서 이 지시사항대로 unknown으로 먼저 cast후 Transaction으로 다시 cast하였습니다.

Reply:
해당 전제는 EthTransaction가 Transaction로 변환 가능하다는 의미인거같은데
이렇게 as unknown으로 하고 as Transaction 했을 경우 제대로된 타입 체크가 되지 못할거같아보입니다.
( as any 하고 as Transaction을 하는것과 같음 )
TransactionDTO.fromEthTransaction 로 하나 만들거나
```
const ethTransaction = await wallet.transfer(...);
return TransactionDTO.fromTransaction({ ...ethTransaction });
```
Transaction 타입 객체를 다시 생성해서 전달하면 어떨까요~?

Reply:
반영완료